bin/version-changes.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/sh
+VERSION=${VERSION:=\$VID\:\$}
+IS_PRE_RELEASE=${IS_PRE_RELEASE:=false}
+
+# replace all instances of $VID:$ with the release version but only when not pre-release.
+if [ $IS_PRE_RELEASE = false ]; then
+	find ./src -name "*.php" -print0 | xargs -0 sed -i '' 's/\$VID:\$/'${VERSION}'/g'
```

Comment:
In order to get it working on GNU/Linux, I needed to remove `''` after `-i` in this line and the others below.
```suggestion
	find ./src -name "*.php" -print0 | xargs -0 sed -i 's/\$VID:\$/'${VERSION}'/g'
```

It looks like syntax of `sed` in Mac and Linux is different. Do you think there is a way to make it work in both systems? Otherwise, could we add some error handling with an explanation or a comment in the code?

Reply:
Ya, that's one problem here is the sed difference between mac and linux.  I'll see if I can detect os and adjust accordingly (my bash fu is limited)

Reply:
Looks like I can detect OS version so I'll add that change.

Reply:
I'm not super happy with what I implemented in 5c81817 because I'd rather substitute just the part of the command needing changed in each expression, but I couldn't figure it out and went over the time I boxed for it so just went with the more verbose solution.  I've got an open request for help from others so if I come across terser (more DRY) syntax I'll add it.  Ready for another review/test @Aljullu

Reply:
@nerrad We've worked around this before without needing to detect OS. Need to use this method:

https://stackoverflow.com/questions/5694228/sed-in-place-flag-that-works-both-on-mac-bsd-and-linux
https://github.com/woocommerce/woocommerce/blob/c63cd802aa149f8f91b33e287aa8d67eb50eab12/bin/package-update.sh#L32

Reply:
Coolio, however, what I've got here avoids the need for the workaround :) (and having to delete the `bak` files).

Reply:
@nerrad Yeah thats a downside, but detecting OS doesn't feel reliable.

As an alternative (and this where we experienced the sed problem before) it looks like `perl` command can do the same thing. See https://github.com/woocommerce/woocommerce-core-to-wordpress-org/blob/master/release.sh#L316

Reply:
> but detecting OS doesn't feel reliable.

In my reading the method I'm using seems to be fairly solid.

> As an alternative (and this where we experienced the sed problem before) it looks like perl command can do the same thing.

Was actually suggested as an alternative in the slack channel I posed a question.  I'll give it a try.